### PR TITLE
Snapshot ref type public

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: api/src/main/java/org/apache/iceberg/SnapshotRefType.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 import java.util.Locale;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 
-enum SnapshotRefType {
+public enum SnapshotRefType {
```
Comment (aokolnychyi) [inline api/src/main/java/org/apache/iceberg/SnapshotRefType.java:24]: I haven't followed the initial implementation but it seems `SnapshotRef` class offers helper boolean methods to check if a ref is a branch or is a tag and hides `SnapshotRefType` on purpose.

Is there a particular reason to make this public, @snazy?

Comment (rdblue) [reply api/src/main/java/org/apache/iceberg/SnapshotRefType.java:24]: Agreed. I don't see a reason to make this public.

**Comment by snazy** [comment]
I needed this, while I was trying (without success for many reasons) to generate JAX-RS code from the spec. Such an approach would need this change.

I'm okay to leave it as it is. Mostly because I could not find at least one openapi-spec-2-JAX-RS code generator that actually works, and therefore use the JSON serialization from `iceberg-core`.

**Comment by aokolnychyi** [comment]
Sounds good, @snazy. Would you mind closing this one and re-opening if needed? Trying to reduce the number of open PRs against our repo.

**Pull request closed without merging**